core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -68,6 +70,8 @@ class KafkaApis(val requestChannel: RequestChannel,
                 val clusterId: String,
                 time: Time) extends Logging {
 
+  val random = new Random()
```

Comment:
good catch. It was added during the initial implementations to generate PIDs before the server side code was ready.